File: 프로그래머스/2/12981. 영어 끝말잇기/영어 끝말잇기.java
import java.util.*;

class Solution {
    public int[] solution(int n, String[] words) {
       
        int cnt = 0;
        String before = "";
        
        HashSet<String> used = new HashSet<>();
        
        for(int i=0; i<words.length; i++){
            if(used.contains(words[i])) break;
            if(before.length() == 0 || before.charAt(before.length()-1) == words[i].charAt(0)){
                before = words[i];
                used.add(words[i]);
                cnt++;
            }else break;
        }
        
        if(cnt == words.length) return new int[] {0,0};
        
        cnt++;
        
        int num = cnt%n;
        if(num == 0) num = n;
        
        int turn = (int)Math.ceil((double)cnt/n);
       
        return new int[] {num, turn};
    }
}

File: 프로그래머스/2/12981. 영어 끝말잇기/영어 끝말잇기.js
function solution(n, words) {
    
    const used = new Map();
    var cnt = 0;
    var before = "";
    
    for(let i=0; i<words.length; i++){
        if(used.has(words[i])) break;
        if(before.length == 0 || before.charAt(before.length-1) == words[i].charAt(0)){
            before = words[i]
            cnt++
            used.set(words[i])
        }else break;
    }
    
    if(cnt == words.length) return [0,0]
    
    cnt++
    
    var num = cnt%n
    if(num == 0) num = n;
    
    var turn = Math.ceil(cnt/n)

    return [num ,turn];
}